tcms/core/tests/test_markdown.py

```diff
@@ -0,0 +1,49 @@
+import unittest
+import bleach
+
+from tcms.core.templatetags.extra_filters import markdown2html
+
+
+class TestMarkdownExtraFilters(unittest.TestCase):
+
+    def test_markdown2html_convert_paragraphs(self):
+        text2html = markdown2html("__*hello!*__")
+        self.assertEqual(bleach.clean(text2html),
+                         "&lt;p&gt;<strong><em>hello!</em></strong>&lt;/p&gt;")
+
+    def test_markdown2html_convert_tables(self):
+        table2html = markdown2html("""|Stx |Desc |\n|----|-----|\n|Head|Title|\n|Text|Text|""")
+        self.assertEqual(bleach.clean(table2html), """&lt;table&gt;

```

Comment:
You have made this PR worse than it was because you are making changes in methods which weren't broken. You have to revert it to what it was before. 

The subject under test is only the markdown2html() function and nothing else. The fact that it uses bleach to escape HTML tags which it doesn't like is a behavior detail. You must be checking the output of markdown2html not the output of bleach.clean.

The test methods which were asserting on the correctness of all of the available markdown extensions were previously correct so revert to their previous version.